src/parachain/issue.ts

```diff
@@ -255,14 +274,17 @@ export class DefaultIssueAPI extends DefaultTransactionAPI implements IssueAPI {
     }
 
     async requestAdvanced(
-        amountsPerVault: Map<AccountId, MonetaryAmount<WrappedCurrency, BitcoinUnit>>,
+        amountsPerVault: Map<string, MonetaryAmount<WrappedCurrency, BitcoinUnit>>,
         atomic: boolean
     ): Promise<Issue[]> {
         const txs = await Promise.all(
             Array.from(amountsPerVault.entries()).map(
                 ([vaultId, amount]) =>
                     new Promise<SubmittableExtrinsic<"promise">>((resolve) => {
-                        this.craftRequestTx(vaultId, amount).then((tx) => resolve(tx));
+                        this.craftRequestTx(
```

Comment:
`craftRequestTx` is doing two network reads in sequence and sends one transaction. In the worst case, this makes this request call very very slow. Can we somehow already submit the griefing collateral as part of making the tx so we get rid of the reads? Can be a future feature though as this is probably not super trivial/requires changes in the UI side of the code as well.

Reply:
If you mean two calls to compute the griefing collateral that's a good point, I changed it to call in parallel